server/src/main/java/io/crate/auth/Credentials.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.auth;
+
+import org.elasticsearch.common.settings.SecureString;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Holder for all authentication methods.
+ * username is used for password, cert and trust methods and null otherwise.
+ * password is used for password method and null otherwise.
+ * jwtToken is used for jwt method and null otherwise.
+ */
+public class Credentials {
+
+    private final String username;
+
+    // Non-final as Postgres protocol might inject password later after creation.
+    private SecureString password;
+
+    private final String jwtToken;
```

Comment:
Any reason to store it in a dedicated variable? Wouldn't `passwordOrToken` be also fine? Do we need this separation?
Beside of that, the token should be also stored in a `SecureString`.

Reply:
>Wouldn't passwordOrToken be also fine?

Then we don't need `Credentials` at all, because we already have Tuple with name/password and can reuse `Tuple's` password field to have semantics "passwordOrToken" and adjust `AuthenticationMethod` signature to `passwordOrToken` as well.

I had something like that on the first iteration and found out that further steps can lead to confusion:

1. In basic auth, username stands for "CrateDB username". In JWT auth, username is not (necessarily) CrateDB username.

Technically, it means that `Tuple.v1()` or `Credentials.username()` would be useless for JWT flow, we have to do unsafe decode + actual user lookup to find out a CrateDB user. Idea was to have a dedicated token field to differentiate - but we can just set `Tuple.name = null` for token and revert this. wdyt?

Reply:
Yes, can use `passwordOrToken`. Was thinking that we need to somehow mark that name needs to be resolved - but we can just use null check for JWT